### ISSUE TITLE
Mapillary Placement Tools

### ISSUE THREAD

**Issue by amahon** [openstreetmap/iD#4426]
Building from SfM reconstructions that Mapillary generates everywhere we have imagery, it's possible to edit map data by placing and editing points in an image.

The pattern, deemed Placement Tools, is demonstrated by [an example](https://bl.ocks.org/oscarlorentzon/0ec42b32dd175ca4cc7518006b888d3a) found in the [Mapillary.JS documentation](https://github.com/mapillary/mapillary-js).

The functionality is expressed in a Mapillary add-in for ArcGIS Pro:
![image](http://blog.mapillary.com/img/2017-07-27-magp-placement.gif)

Currently, Mapillary.JS only supports placing and editing points in imagery. Forthcoming, but unscheduled, is support for linestrings and polygons.

Also, **one caveat**, Placement tools are only as good as Mapillary's underlying data is. In places like Amsterdam, where Mapillary has high quality imagery with very good GPS data, Placement Tools work fantastically. In areas where imagery/gps data is not as good, Placement Tools will not perform as well.

Let's think about how we can bring this functionality into iD Editor to allow OSM editors to place and edit nodes via Mapillary Imagery.

**Comment by amahon** [comment]
@bhousel - let me know what you think and how I can help!

**Comment by bhousel** [comment]
Very cool, this will make mapping objects so easy 👍

**Comment by planemad** [comment]
Having this will completely unleash a new wave of detailed editing on OSM. I'm going to immediately use it for tree mapping when its out!🌲🌲🌲

**Comment by tordans** [comment]
@amahon I saw [a presentation about this (YouTube)](https://youtu.be/uK_wYFTtj08?t=10m4s) and would love to see this feature live. 
What is the status? 
What is the timeline?
Who is writing the code to get it into iD?

**Comment by amahon** [comment]
@tordans - thanks for your questions!

Right now, we're working on implementing an API that will allow iD editor to access the detection data. I don't have an up to date timeline that I can share. Last discussed, which is a while ago now, Mapbox would implement the new features in iD -- since so much time has passed, I need to loop back and assess priorities with @bhousel. Will post an update here.

Let me know if I can help with anything else!

**Comment by tordans** [comment]
@amahon any news about this?

**Comment by tordans** [comment]
I noticed a surge of attention for this feature (outside of iD). For for reference, a few links: 

- 2019-02 Blogpost ["Deriving Data from Imagery: An Intro to Deriviste and OpenStreetMap"](https://blog.mapillary.com/update/2019/02/19/intro-to-deriviste-and-openstreetmap.html) on the Mapillary Blog and [corresponding Tweet](https://twitter.com/mapillary/status/1097784839732973568)
- 2019-02 Update to Deriviste ([Tweet](https://twitter.com/richardf/status/1091267844430532608))
- 2018-10 [Mapillary-Tweet](https://twitter.com/mapillary/status/1053242513526538240) about the Deriviste-Announcement

**Comment by todrobbins** [comment]
@amahon the people demand SfM tooling! 🛠

**Comment by cbeddow** [comment]
Yes, Deriviste is a good example of how it can work. It also is in the go2mapillary plugin for QGIS, in case another example is needed. I wouldn't personally develop this for iD editor as Deriviste is way more simple for me to work with and OSM iD usually over my head, but if anybody is looking to add it I'm totally open to advise!
